### ISSUE TITLE
[FeatureRequest] Print absolute file path instead of relative

### ISSUE THREAD

**Issue by webzepter** [palantir/tslint#1794]
### Feature request

Hello, I'm using tslint with Webstorm and it would be nice to have tappable linter output. Currently tslint can print only relative file path, but Webstorm requires absolute path :(
Actual result:
<img width="856" alt="screen shot 2016-11-28 at 15 22 19" src="https://cloud.githubusercontent.com/assets/4348222/20669747/7fee08c2-b57e-11e6-88b9-23b31d119465.png">

Expected result:
<img width="1129" alt="screen shot 2016-11-28 at 15 18 27" src="https://cloud.githubusercontent.com/assets/4348222/20669645/007e1ba4-b57e-11e6-89ae-0ea896723e1b.png">

**Comment by nchen63** [comment]
You could probably write this pretty easily using a [custom formatter](https://palantir.github.io/tslint/develop/custom-formatters/)

**Comment by tdsmithATabc** [comment]
Haha I came here to investigate the opposite problem (I only see absolute paths which are cluttering up the console). Maybe we should just switch machines, eh?

**Comment by timfish** [comment]
The output I see is even worse as its mixed! 

Some paths are relative and some are absolute. Obfuscated examples below

```shell
C:/Users/tim/Documents/....ts[16, 19]: Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.
C:/Users/tim/Documents/.....ts[11, 14]: Use an interface instead of a type literal.
src/resources/....ts[14, 31]: Identifier 'lenk' is never reassigned; use 'const' instead of 'let'.
src/resources/....ts[79, 5]: Expected a 'for-of' loop instead of a 'for' loop with this simple iteration
src/resources/...ts[86, 5]: Expected a 'for-of' loop instead of a 'for' loop with this 
```

**Comment by tdsmithATabc** [comment]
If it's useful to add, I _also_ have mixed relative and absolute paths (on v4.0.2). At first I thought it was because of my set up (faking the application folder as an additional `app` node module so I can do package-relative imports) but it occurs regardless of whether I'm using that approach or not.

That said it hasn't impacted me personally because we use unix tools to filter and sort the output anyway and enough of the path remains to still identify everything easily.

**Comment by timfish** [comment]
I'm using vscode and the console output pattern matcher has to define whether the paths are relative or absolute. This means that whichever way I configure it, some of the paths are not recognised.

**Comment by IllusionMH** [comment]
It is interesting how do you invoke TSLint so you have mixed path? Do you have any symlinks there?

**Comment by timfish** [comment]
I'm using:

`tslint --config ./src/tslint.json --project tsconfig.json --type-check --force`

All plain old files with no symlinks

`tslint.json`
```json
{
  "extends": "tslint:latest",
  "rules": {
    "quotemark": [
      true,
      "single",
      "avoid-escape"
    ],
    "ordered-imports": [
      false
    ],
    "max-line-length": [
      false
    ],
    "max-classes-per-file": [
      false
    ],
    "trailing-comma": [
      false
    ],
    "no-trailing-whitespace": false,
    "member-ordering": [
      false
    ],
    "object-literal-sort-keys": false,
    "no-empty": false,
    "object-literal-key-quotes": [
      false
    ],
    "no-unused-variable": true,
    "radix": false,
    "no-null-keyword": true,
    "no-angle-bracket-type-assertion": true,
    "triple-equals": [
      true,
      "allow-undefined-check"
    ],
    "arrow-parens": false,
    "array-type": [
      true,
      "array"
    ],
    "prefer-const": true,
    "prefer-for-of": false
  }
}
```

**Comment by tdsmithATabc** [comment]
No symlinks here. My command is:
```shell
tslint --type-check --config config/tslint.json --project tsconfig.json
```

I don't think there's anything notable in my tslint config (just enabling rules and setting the custom directory). For tsconfig, I'm using `commonjs` modules with `node` resolution.

As previously mentioned, I use some module-style imports to access local files like so:
```
import { AlertService } from "app/common/alert.service";
```
But I've experimented with traditional relative imports and that seems to have no bearing on the mixed paths issue. I can try to double-check this again soon though.

**Comment by timfish** [comment]
Here are some slightly less obfuscated warnings, this time with msbuild formatter. 

Points to note:
- It seems unrelated to the rule which is failing.
- It always switches from absolute to relative somewhere towards the end.
- Its repeatable. If I run the command again, I get the same output

```shell
c:/Users/tim/Documents/Project/src/components/confirm-dialog.ts(25,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/components/confirm-dialog.ts(31,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/components/confirm-dialog.ts(37,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/components/confirm-dialog.ts(43,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/devices/controls/edit-channel/dialog.ts(156,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/devices/controls/edit-channel/dialog.ts(162,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/devices/controls/edit-channel/dialog.ts(168,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/devices/controls/edit-channel/dialog.ts(174,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/utils/color/color-convert.ts(31,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
c:/Users/tim/Documents/Project/src/devices/dal/device-configure.ts(66,13): warning typeofCompare: typeof must be compared to correct value
src/graph/components/axis/x-axis.ts(42,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
src/graph/components/axis/y-axis.ts(34,1): warning maxClassesPerFile: A maximum of 1 class per file is allowed
```

**Comment by colinbate** [comment]
I've also started experiencing mixed relative and absolute paths in the output, but I think I know what caused it (roughly). All my paths were relative until I added some rules which required type checking, then I switched to using `--project path/to/tsconfig.json` and `--type-check`. Prior to that, I was just using a file glob: `path/to/**/*.ts`.

This is using Windows 7. I just confirmed that removing those rules and going back to the glob instead of the project that they all return to relative.

Mine switch back and forth between absolute and relative a couple times. The rules don't seem to matter, but the paths for a single file tend to always be the same style.

**Comment by Nefarion** [comment]
For all the VSCode Users, you can replace the problemMatcher in your task with this one, it will distinguish between absolute (C:/, D:/,...) Windows Paths, and relative ones (src/app/...)

`"problemMatcher": [{
            "owner": "tslint",
            "severity": "warning",
            "fileLocation": "absolute",
            "pattern": {
                "regexp": "^(\\S[:][\\/]\\S.*)\\[(\\d+), (\\d+)\\]:\\s+(.*)$",
                "file": 1,
                "line": 2,
                "column": 3,
                "message": 4
            }
        },
        {
            "owner": "tslint",
            "severity": "warning",
            "fileLocation": ["relative", "${workspaceRoot}"],
            "pattern": {
                "regexp": "^(\\S[^:][^\\/]\\S.*)\\[(\\d+), (\\d+)\\]:\\s+(.*)$",
                "file": 1,
                "line": 2,
                "column": 3,
                "message": 4
            }
        }`

Now i get warnings in the Problems Tab!

**Comment by rweng** [comment]
For osx/linux users:

```json
"problemMatcher": [
                {
                    "base": "$tslint5",
                    "fileLocation": "absolute",
                    "pattern": {
                        "regexp": "^(WARNING|ERROR):(\\s+\\(/\\S*\\))?\\s+(/\\S.*)\\[(\\d+), (\\d+)\\]:\\s+(.*)$",
                        "severity": 1,
                        "file": 3,
                        "line": 4,
                        "column": 5,
                        "message": 6
                    }
                },
                {
                    "base": "$tslint5",
                    "fileLocation": "relative",
                    "pattern": {
                        "regexp": "^(WARNING|ERROR):(\\s+\\(/\\S*\\))?\\s+([^/]\\S.*)\\[(\\d+), (\\d+)\\]:\\s+(.*)$",
                        "severity": 1,
                        "file": 3,
                        "line": 4,
                        "column": 5,
                        "message": 6
                    }
                }
            ]
```
